server/extension/extension.go

```diff
@@ -447,6 +513,32 @@ func (m *Manager) authorize(ctx context.Context, rr *RequestResources, extName s
 	return app, nil
 }
 
+func findProxy(proxyByCluster map[string]*httputil.ReverseProxy, extName string, dest v1alpha1.ApplicationDestination) (*httputil.ReverseProxy, error) {
+
+	key := proxyByClusterKey(extName, "")
+	if proxy, found := proxyByCluster[key]; found {
+		return proxy, nil
+	}
+
+	if dest.Name != "" && dest.Server != "" {
+		return nil, fmt.Errorf("invalid application: both destination name and destination server are provided")
+	}
+	if dest.Name != "" {
+		key = proxyByClusterKey(extName, dest.Name)
```

Comment:
This allows the name to match a server URL, because the key does not differentiate between server URL and name.

I'd recommend using this as the key instead of an extension name -> server/url map.

```go
type key struct {
	extension string
	clusterServer string
	clusterName string
}
```

You'd still populate the map with duplicate entries. But in this case, name will only ever be used to match name, and server will only ever be used to match server.

Reply:
fixed